main.cc

```diff
@@ -146,6 +146,62 @@ read_config(bpo::variables_map& opts, db::config& cfg) {
         return make_exception_future<>(ep);
     });
 }
+
+class sighup_handler {
+    bpo::variables_map& _opts;
+    db::config& _cfg;
+    condition_variable _cond;
+    bool _pending = false; // if asked to reread while already reading
+    bool _stopping = false;
+    bool _stopped = false;
+    future<> _done = do_work();
```

Comment:
A blurb would be very helpful here.

Reply:
It's a standard Seastar pattern, but okay.